Tests/HummingbirdTests/ApplicationTests.swift

```diff
@@ -893,6 +893,72 @@ final class ApplicationTests: XCTestCase {
             }
         }
     }
+
+    /// Test consumeWithInboundHandler after having collected the Request body
+    @available(macOS 15, iOS 18, tvOS 18, *)
+    func testConsumeWithInboundHandlerAfterCollect() async throws {
+        let router = Router()
+        router.post("streaming") { request, context -> Response in
+            var request = request
+            _ = try await request.collectBody(upTo: .max)
+            let request2 = request
+            return Response(
+                status: .ok,
+                body: .init { writer in
+                    try await request2.body.consumeWithInboundCloseHandler { body in
+                        try await writer.write(body)
+                    } onInboundClosed: {
```

Comment:
Same here, shouldn't we assert whether or not it gets hit?

Reply:
No because the client never closes the connection